### PR TITLE
Admin ui to manage authz

### PR DESCRIPTION
**Pull request by sindrig**
- [x] Understand review feedback
- [x] Fix docker-compose.yml: changed functions mount from read-only to read-write and added `npm ci && npm run build` before starting the emulator
- [x] Fix onUserCreate.ts: add `!Array.isArray` guard in `isInvitationRecord` to reject arrays as location objects
- [x] Fix onUserCreate.ts: additive merging — only copy `true` values so a later invitation cannot flip a previously granted `true` to `false`
- [x] Fix adminWrite.ts: refactored `assertValidEmail` into `validateAndNormalizeEmail` that both validates and returns the normalized (trimmed/lowercased) email, eliminating the need for a separate `.trim().toLowerCase()` call in the handler
- [x] Add test: invitations not deleted when auth write fails
- [x] Add test: merges locations additively (true always wins over false)
- [x] Add test: email with surrounding whitespace is accepted and stored normalized
- [x] All 30 tests pass, TypeScript build clean, no CodeQL alerts

### CODE REVIEW

File: .opencode/plans/004-react-router-integration.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+pnpm add react-router-dom
+```
+
+### New file: `clock/src/router.tsx`
+
+Define the app's routes:
+
+```
+/          -> existing App component (all current behavior)
+/admin     -> AdminPortal component (new, see spec 006)
```
Comment (sindrig) [inline .opencode/plans/004-react-router-integration.md]: this will require some changes to s3 website hosting or cloudfront so that /whateverpath renders index.html - making sure cache busting is in effect

File: .opencode/plans/006-admin-portal-ui.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+| Screens | RTDB `auth/{uid}` | Checkboxes for each location from `locations/` |
+| Last Sign-In | Firebase Auth | Last login timestamp |
+| Actions | -- | Edit button to modify screen access |
+
+**Interactions**:
+- Click "Breyta" (Edit) on a user row -> opens a modal/drawer with checkbox list of all locations
+- Toggle location checkboxes -> on save, calls `adminWrite` Cloud Function with `setUserLocations` action
+- Changes appear in real-time because the admin is subscribed to `auth/` via `onValue`
+
+**Empty state**: "Engir notendur skraddir" (No registered users) if the user list is empty.
```
Comment (sindrig) [inline .opencode/plans/006-admin-portal-ui.md]:
```suggestion
**Empty state**: "Engir notendur skráðir" (No registered users) if the user list is empty.
```

File: .opencode/plans/006-admin-portal-ui.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+- `clock/src/firebase.ts` -- export `functions` instance (see spec 003)
+- `clock/src/controller/Controller.tsx` -- add red admin button (see spec 004)
+
+## Admin button in Controller.tsx
+
+When `isAuthenticated && !listenPrefix && isAdmin`, render above the screen selector buttons:
+
+```tsx
+<Link to="/admin" className="admin-portal-button">
+  Stjornbord
```
Comment (sindrig) [inline .opencode/plans/006-admin-portal-ui.md]:
```suggestion
  Stjórnbord
```

File: .opencode/plans/007-cicd-cloud-functions.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        tools-version: 11.9.0
+        firebase_token: ${{ secrets.FIREBASE_TOKEN }}
+
+    - name: Deploy Cloud Functions
+      run: firebase deploy --only functions -P vikes-match-clock-firebase --token $FIREBASE_TOKEN
+      env:
+        FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+```
+
+### Staging deployment
```
Comment (sindrig) [inline .opencode/plans/007-cicd-cloud-functions.md]: extract the logic of deploying cloud functions to a reusable workflow to avoid having it duplicated in the main workflow
